calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -305,3 +306,118 @@ def test_validate_arguments(self, m_sys_exit):
             else:
                 assert m_sys_exit.called
             m_sys_exit.reset_mock()
+
+
+class TestUtils(unittest.TestCase):
+
+    @patch('sys.exit', autospec=True)
+    def test_check_ip_version(self, m_sys_exit):
+        """
+        Test check_ip_version function in calicoctl utils
+        """
+        # Set up arguments
+        cases = (
+            ('1.2.3.4', 'v4', IPAddress, True, 1),
+            ('1.2.3.4', 'v6', IPAddress, False, 2),
+            ('1.2.3.4', 'v4', IPNetwork, True, 3),
+            ('1.2.3.0/24', 'v4', IPAddress, False, 4),
+            ('1.2.3.0/24', 'v4', IPNetwork, True, 5),
+            ('aa:bb::ff', 'v4', IPAddress, False, 6),
+            ('aa:bb::ff', 'v6', IPAddress, True, 7)
+        )
+
+        # Call method under test for each test case
+        # Assert that method exits on bad input
+        for case in cases:
```

Comment:
You can actually merge this step with the step below.  Just replace the `case` here with the tuple definition and it will understand what you mean.  This goes for the other tests too (i'll note them below)